src/Raven.Server/Documents/PeriodicBackup/Retention/AzureRetentionPolicyRunner.cs

```diff
@@ -18,22 +25,75 @@ public AzureRetentionPolicyRunner(RetentionPolicyBaseParameters parameters, Rave
 
         protected override GetFoldersResult GetSortedFolders()
         {
-            throw new NotSupportedException();
+            var prefix = $"{_client.RemoteFolderName}{Delimiter}";
+            var result = _client.ListBlobs(prefix, Delimiter, listFolders: true, marker: _folderContinuationToken).Result;
+            _folderContinuationToken = result.NextMarker;
+
+            return new GetFoldersResult
+            {
+                List = result.ListBlob.Select(x => x.Name).OrderBy(x => x).ToList(),
+                HasMore = result.NextMarker != null
+            };
         }
 
         protected override string GetFolderName(string folderPath)
         {
-            throw new NotSupportedException();
+            return folderPath.Substring(0, folderPath.Length - 1);
         }
 
         protected override GetBackupFolderFilesResult GetBackupFilesInFolder(string folder, DateTime startDateOfRetentionRange)
         {
-            throw new NotSupportedException();
+            var backupFiles = new GetBackupFolderFilesResult();
+            var blobs = _client.ListBlobs(folder, delimiter: null, listFolders: false, maxResult: 2).Result;
+            backupFiles.FirstFile = blobs.ListBlob?.Select(x => x.Name).FirstOrDefault();
+
+            blobs = _client.ListBlobs(folder, delimiter: null, listFolders: false).Result;
```

Comment:
We can skip this if `backupFiles.FirstFile == null`

Reply:
done